```
src\libraries\System.Private.CoreLib\src\System\TimeZoneInfo.cs#L1935
src\libraries\System.Private.CoreLib\src\System\TimeZoneInfo.cs(1935,22): error CS7036: (NETCORE_ENGINEERING_TELEMETRY=Build) There is no argument given that corresponds to the required formal parameter 'e' of 'TimeZoneInfo.TryGetTimeZoneFromLocalMachine(string, bool, out TimeZoneInfo?, out Exception?, TimeZoneInfo.CachedData)'
```

--- a/src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Win32.cs
+++ b/src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Win32.cs
@@ -773,8 +773,8 @@ namespace System
         /// </summary>
         private static unsafe string GetFileMuiPath(string filePath, CultureInfo cultureInfo)
         {
-            char* fileMuiPath = stackalloc char[Interop.Kernel32.MAX_PATH];
-            char* language = stackalloc char[Interop.Kernel32.LOCALE_NAME_MAX_LENGTH];
+            char* fileMuiPath = stackalloc char[Interop.Kernel32.MAX_PATH + 1];
+            char* language = stackalloc char[Interop.Kernel32.LOCALE_NAME_MAX_LENGTH + 1];
             uint fileMuiPathLength = Interop.Kernel32.MAX_PATH;
             uint languageLength = Interop.Kernel32.LOCALE_NAME_MAX_LENGTH;
             ulong enumerator = 0;
@@ -804,6 +804,7 @@ namespace System
 
                     if (succeeded)
                     {
+                        fileMuiPath[Interop.Kernel32.MAX_PATH] = '\0';
                         return new string(fileMuiPath);
                     }
 
@@ -812,9 +813,11 @@ namespace System
                 }
 
                 // Lookup succeeded.  Check for exact match to the desired culture.
+                language[Interop.Kernel32.LOCALE_NAME_MAX_LENGTH] = '\0';
                 var lang = new string(language);
                 if (string.Equals(lang, cultureInfo.Name, StringComparison.OrdinalIgnoreCase))
                 {
+                    fileMuiPath[Interop.Kernel32.MAX_PATH] = '\0';
                     return new string(fileMuiPath);
                 }
 
@@ -824,6 +827,7 @@ namespace System
                 {
                     if (ci.Parent.Name.Equals(cultureInfo.Name, StringComparison.OrdinalIgnoreCase))
                     {
+                        fileMuiPath[Interop.Kernel32.MAX_PATH] = '\0';
                         return new string(fileMuiPath);
                     }
 

--- a/src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Win32.cs
+++ b/src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Win32.cs
@@ -995,7 +995,7 @@ namespace System
         /// Helper function that takes a string representing a time_zone_name registry key name
         /// and returns a TimeZoneInfo instance.
         /// </summary>
-        private static TimeZoneInfoResult GetTimeZoneFromLocalMachine(string id, out TimeZoneInfo? value, out Exception? e)
+        private static TimeZoneInfoResult TryGetTimeZoneFromLocalMachine(string id, out TimeZoneInfo? value, out Exception? e)
         {
             e = null;
 

--- a/src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs
+++ b/src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs
@@ -2723,7 +2723,8 @@ namespace System.Tests
             }).Dispose();
         }
 
-        private static bool CanTestWindowsNlsDisplayNames => s_isWindows && RemoteExecutor.IsSupported && WindowsUILanguageHelper.GetInstalledWin32CulturesWithUniqueLanguages().Length > 1;
+        private static readonly CultureInfo[] s_CulturesForWindowsNlsDisplayNamesTest = WindowsUILanguageHelper.GetInstalledWin32CulturesWithUniqueLanguages();
+        private static bool CanTestWindowsNlsDisplayNames => RemoteExecutor.IsSupported && s_CulturesForWindowsNlsDisplayNamesTest.Length > 1;
 
         [PlatformSpecific(TestPlatforms.Windows)]
         [ConditionalFact(nameof(CanTestWindowsNlsDisplayNames))]
@@ -2731,7 +2732,7 @@ namespace System.Tests
         {
             RemoteExecutor.Invoke(() =>
             {
-                var cultures = WindowsUILanguageHelper.GetInstalledWin32CulturesWithUniqueLanguages();
+                CultureInfo[] cultures = s_CulturesForWindowsNlsDisplayNamesTest;
 
                 CultureInfo.CurrentUICulture = cultures[0];
                 TimeZoneInfo.ClearCachedData();
@@ -2741,9 +2742,9 @@ namespace System.Tests
                 TimeZoneInfo.ClearCachedData();
                 TimeZoneInfo tz2 = TimeZoneInfo.FindSystemTimeZoneById(s_strPacific);
 
-                Assert.True(tz1.StandardName != tz2.DisplayName, $"The display name '{tz1.DisplayName}' should be different between {cultures[0].Name} and {cultures[1].Name}.");
+                Assert.True(tz1.DisplayName != tz2.DisplayName, $"The display name '{tz1.DisplayName}' should be different between {cultures[0].Name} and {cultures[1].Name}.");
                 Assert.True(tz1.StandardName != tz2.StandardName, $"The standard name '{tz1.StandardName}' should be different between {cultures[0].Name} and {cultures[1].Name}.");
-                Assert.True(tz1.StandardName != tz2.DaylightName, $"The daylight name '{tz1.DaylightName}' should be different between {cultures[0].Name} and {cultures[1].Name}.");
+                Assert.True(tz1.DaylightName != tz2.DaylightName, $"The daylight name '{tz1.DaylightName}' should be different between {cultures[0].Name} and {cultures[1].Name}.");
             }).Dispose();
         }
 
@@ -3009,7 +3010,7 @@ namespace System.Tests
         }
 
         //  This helper class is used to retrieve information about installed OS languages from Windows.
-        //  It should only be using in Windows platform-specific tests.
+        //  Its methods returns empty when run on non-Windows platforms.
         private static class WindowsUILanguageHelper
         {
             public static CultureInfo[] GetInstalledWin32CulturesWithUniqueLanguages() =>
@@ -3020,6 +3021,11 @@ namespace System.Tests
 
             public static CultureInfo[] GetInstalledWin32Cultures()
             {
+                if (!OperatingSystem.IsWindows())
+                {
+                    return new CultureInfo[0];
+                }
+
                 var context = new EnumContext();
                 EnumUILanguagesProc proc = EnumUiLanguagesCallback;
 

